dispatch enter/exit events on TextTrackCue/VTTCue with no track

karlcow opened the pull request:
#### 8956674a34a3f1cd0ddc7c0ff56f3d959cf66ae6
<pre>
dispatch enter/exit events on TextTrackCue/VTTCue with no track
<a href="https://bugs.webkit.org/show_bug.cgi?id=298227">https://bugs.webkit.org/show_bug.cgi?id=298227</a>
<a href="https://rdar.apple.com/160195643">rdar://160195643</a>

Reviewed by Alicia Boya Garcia and Eric Carlson.

This removes the constraint to have a track associated for dispatching
events. Then as it inherits from the normal dispatchEvent, it doesn&apos;t
require its own declaration.

* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/interfaces/TextTrackCue/onenter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/interfaces/TextTrackCue/onexit-expected.txt:
* Source/WebCore/html/track/TextTrackCue.cpp:
(WebCore::TextTrackCue::dispatchEvent): Deleted
* Source/WebCore/html/track/TextTrackCue.h:
(WebCore::TextTrackCue::dispatchEvent): Deleted

Canonical link: <a href="https://commits.webkit.org/301836@main">https://commits.webkit.org/301836@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d2fd61c5fb1fd42b3e6ae69126c543dc8a2a2782

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126297 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45951 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36790 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133146 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78053 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/772a7f75-af3c-4f8a-98d0-cd5f3a2daa09) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128168 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46614 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54495 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96152 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64258 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4298aea3-ef13-4ab7-a9b0-37a88c6b203a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129245 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37290 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112973 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77259 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/83a0c559-2f42-4968-b3d1-3ee63fc704c2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36190 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31149 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76533 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107075 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31388 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136654 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53037 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105260 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53501 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109216 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104948 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26769 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49798 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28134 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50420 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52939 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58765 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52251 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55594 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53970 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

ntrrgc commented on Source/WebCore/html/track/TextTrackCue.cpp:
```diff
@@ -338,8 +338,8 @@ void TextTrackCue::setPauseOnExit(bool value)
-        return;
+    // if (!track() || track()->mode() == TextTrack::Mode::Disabled)
+    //     return;
 
```
We shouldn't leave commented out code in WebKit.

karlcow replied:
yes this was just for testing and see if anything was breaking. see my comment above.

karlcow replied:
fixed

ntrrgc commented on Source/WebCore/html/track/TextTrackCue.cpp:
```diff
@@ -338,8 +338,8 @@ void TextTrackCue::setPauseOnExit(bool value)
+    // if (!track() || track()->mode() == TextTrack::Mode::Disabled)
+    //     return;
 
     EventTarget::dispatchEvent(event);
```
Since now this method just calls the parent method, we can remove `TextTrackCue::dispatchEvent(Event&)` from both the .h and the .cpp files.

karlcow replied:
done

ntrrgc reviewed (changes requested):
Other than the style nits, I agree with the rationale in the bugzilla issue and I think the change makes sense in the abstract.

I do wonder however why the filter existed in the first place. @eric-carlson Do you have any idea?

karlcow committed:
dispatch enter/exit events on TextTrackCue/VTTCue with no track
https://bugs.webkit.org/show_bug.cgi?id=298227
rdar://160195643

Reviewed by Alicia Boya Garcia and Eric Carlson.

This removes the constraint to have a track associated for dispatching
events. Then as it inherits from the normal dispatchEvent, it doesn't
require its own declaration.

* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/interfaces/TextTrackCue/onenter-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/interfaces/TextTrackCue/onexit-expected.txt:
* Source/WebCore/html/track/TextTrackCue.cpp:
(WebCore::TextTrackCue::dispatchEvent): Deleted
* Source/WebCore/html/track/TextTrackCue.h:
(WebCore::TextTrackCue::dispatchEvent): Deleted

Canonical link: https://commits.webkit.org/301836@main
